lib/tok.js

```diff
@@ -362,6 +365,26 @@
       this.lastType = type;
       return type;
     },
+
+    /** Add line information to tokens. The last token
+     * on a line of code is a carriage return (⏎). Single
+     * line or multi-line comments are retrieved as a single
+     * comment token.
+     * */
+    addLineCount: function(token, type, lastValue){
+      token['line'] = this.lineCounter;
+      if(type !== WHITE){
+        // This token is anything other than a line terminator
+        // and a single or multiple comment line.
+        return;
+      }
+
+      var carriageMatches = lastValue.match(/[\r\n]/g);
```

Comment:
I think you missed the exotic PS and LS chars (`0x2028` and `0x2029`) which are formally considered newlines in JS.

How would this count multi-line comments with multiple newlines? And before you address this please consider that JS considers the combo `crlf` as a single newline for the sake of line reporting.

Reply:
Yes I would now have to change it. My understanding was that for instance n lines would have at least n-1 line separators.